### ISSUE TITLE
Emulate LATERAL (TableImpl | JoinTable) where not supported

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14988]
Some RDBMS support referencing ordinary tables in `LATERAL` or `APPLY`, e.g. SQL Server:

```sql
select *
from t_book
cross apply t_author
where t_book.author_id = t_author.id
```

The same isn't possible in PostgreSQL, for example:

```sql
select *
from t_book, lateral (t_author)
where t_book.author_id = t_author.id
```

> ERROR: syntax error at or near "t_author"

We have to emulate the syntax as follows:

```sql
select *
from t_book, lateral (select * from t_author) as t_author
where t_book.author_id = t_author.id
```

Alternatively, we could perhaps just omit `LATERAL` entirely, as it serves no purpose.

```sql
select *
from t_book, t_author
where t_book.author_id = t_author.id
```

### Support matrix:

- [x] Db2: No support for `LATERAL (table ref)`
- [x] Firebird: No support for `LATERAL (table ref)`
- [x] MySQL: No support for `LATERAL (table ref)`
- [x] Oracle: Native support for `APPLY table ref` but not `LATERAL (table ref)`
- [x] Snowflake: No support for `LATERAL (table ref)`
- [x] SQL Server: Native support for `APPLY table ref`
- [x] Trino: No support for `LATERAL (table ref)`

Or, in short:

- `APPLY` supports this natively
- `LATERAL` does not

The main use-case for this improvement is to enable path joins in `APPLY` and `LATERAL`, see: https://github.com/jOOQ/jOOQ/issues/14985

**Comment by lukaseder** [comment]
This change is a part of https://github.com/jOOQ/jOOQ/issues/14985, and depends on some new internal infrastructure, so it won't be backported. It's unlikely many people expect this to work anyway, given that `A CROSS APPLY B` is effectively just `A CROSS JOIN B`.

**Comment by lukaseder** [comment]
Implemented as part of #14985
